app/services/invoices/calculate_fees_service.rb

```diff
@@ -235,5 +248,25 @@ def calculate_boundaries(subscription)
     def not_in_finalizing_process?
       invoice.draft? && context != :finalize
     end
+
+    def coupon_fees(applied_coupon)
+      invoice
+        .fees
+        .joins(subscription: :plan)
+        .where(plan: { id: applied_coupon.coupon.coupon_plans.select(:plan_id) })
+    end
+
+    def coupon_base_amount_cents(coupon_related_fees:)
+      fee_amounts = coupon_related_fees.select(:amount_cents, :vat_amount_cents)
+
+      fees_amount_cents = fee_amounts.sum(&:amount_cents)
+      fees_vat_amount_cents = fee_amounts.sum(&:vat_amount_cents)
+
+      total_fees_amount_cents = fees_amount_cents + fees_vat_amount_cents
+
+      # In some cases when credit note is already applied sum from above
+      # can be greater than invoice total_amount_cents
+      total_fees_amount_cents > invoice.total_amount_cents ? invoice.total_amount_cents : total_fees_amount_cents
```

Comment:
Style/TernaryParentheses: Use parentheses for ternary expressions with complex conditions.